topology/probes/istio/virtualservice.go

```diff
@@ -88,20 +89,20 @@ func virtualServicePodMetadata(a, b interface{}, typeA, typeB, manager string) g
 	if pod.Labels["app"] == "" {
 		return m
 	}
-	weight := 0
 	for _, http := range vsSpec.HTTP {
+		if http.Match != nil {
```

Comment:
Why do you skip when `http.Match` is set? I found examples in which both `match` and `routes` are set:

https://istio.io/docs/reference/config/networking/v1alpha3/virtual-service/

```
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: reviews-route
  namespace: foo
spec:
  hosts:
  - reviews # interpreted as reviews.foo.svc.cluster.local
  http:
  - match:
    - uri:
        prefix: "/wpcatalog"
    - uri:
        prefix: "/consumercatalog"
    rewrite:
      uri: "/newcatalog"
    route:
    - destination:
        host: reviews # interpreted as reviews.foo.svc.cluster.local
        subset: v2
  - route:
    - destination:
        host: reviews # interpreted as reviews.foo.svc.cluster.local
        subset: v1
```

Reply:
A virtualservice that is configured properly is ending with a default rule, which is intended for a traffic that doesn't match any former conditional rule. As we discussed before, presenting weights for conditional rules doesn't make sense as we don't know the traffic origin beforehand. Thus, we should verify that a rule is the default one before adding its weights to the metadata field.

Reply:
Ok - please add a comment stating the above (as this is not apparent from reading the code).

Reply:
DONE